### DISCUSSION TITLE
SSL and non-SSL communication at the same time?

### DISCUSSION THREAD

**Discussion by MichaelUray** [rabbitmq/rabbitmq-server#4494] [Questions] [answered]
I noticed, that RabbitMQ disables the port for the non-encrypted communication as soon as I assign certs via docker-compose.
It also automatically assigned my certs to the management interface, even I only set `RABBITMQ_SSL_x` and I kept the variables `RABBITMQ_MANAGEMENT_SSL_x` empty.

I would like to use assigned certs to encrypt the MQTT communication, but not the management and also want to allow unencrypted MQTT communication to the port (1883) even the TLS option for port 8883 is enabled.

- Is there a way to allow SSL and non-SSL communication at the same time?
- Is there a way to use SSL only for MQTT, but not for the management?

**Comment by lukebakken** [comment, marked as answer]
> I noticed, that RabbitMQ disables the port for the non-encrypted communication as soon as I assign certs via docker-compose.

This is a docker or configuration issue, not something that RabbitMQ does automatically.

> Is there a way to allow SSL and non-SSL communication at the same time?

Yes, see here -

https://github.com/lukebakken/mqtt-client-cert-example/blob/main/rabbitmq.conf.in#L14-L15

> Is there a way to use SSL only for MQTT, but not for the management?

Yes. Just don't configure TLS for `management`. If you don't do any `management` configuration the default non-TLS HTTP port 1f 15672 will be used.
